cmd/nerdctl/build.go

```diff
@@ -73,22 +74,59 @@ func newBuildCommand() *cobra.Command {
 	return buildCommand
 }
 
+func getBuildkitHost(cmd *cobra.Command) (string, error) {
+	if cmd.Flags().Changed("buildkit-host") {
+		// If address is explicitly specified, use it.
+		buildkitHost, err := cmd.Flags().GetString("buildkit-host")
+		if err != nil {
+			return "", err
+		}
+		if err := buildkitutil.PingBKDaemon(buildkitHost); err != nil {
+			return "", err
+		}
+		return buildkitHost, nil
+	}
+	ns, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return "", err
+	}
+	return buildkitutil.GetBuildkitHost(ns)
+}
+
+func isImageSharable(buildkitHost string, namespace, uuid string) (bool, error) {
+	labels, err := buildkitutil.GetWorkerLabels(buildkitHost)
+	if err != nil {
+		return false, err
+	}
+	logrus.Debugf("worker labels: %+v", labels)
+	executor, ok := labels["org.mobyproject.buildkit.worker.executor"]
+	if !ok {
+		return false, nil
+	}
+	containerdUUID, ok := labels["org.mobyproject.buildkit.worker.containerd.uuid"]
+	if !ok {
+		return false, nil
+	}
+	containerdNamespace, ok := labels["org.mobyproject.buildkit.worker.containerd.namespace"]
+	if !ok {
+		return false, nil
+	}
+	return executor == "containerd" && containerdUUID == uuid && containerdNamespace == namespace, nil
```

Comment:
Maybe we should compare the snapshotter too?
To avoid some potential issues with unpacked images.

Reply:
cc @ktock

Reply:
SGTM